### PR TITLE
Uploading a malformed pipeline yaml results in a blank page

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: #1522  

## Description
#1522 
Error page is being displayed instead of redirecting it to blank page
Screenshot from 2023-07-20 12-02-59

## How Has This Been Tested?
Import this pipleline yaml file
[TestPipeline01-modified.yaml.txt](https://github.com/opendatahub-io/odh-dashboard/files/12102879/TestPipeline01-modified.yaml.txt)
and check for the error page 

## Test Impact
no test coverage

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by pnaik1** [comment]
@kywalker-rh can you please take a look at this?

**Review by manaswinidas — COMMENTED** [review]
Not sure whether we need the `Graph` and `YAML` navigation tabs anymore. We need UX inputs in this. 

I referred to [the screenshot in this PR](https://github.com/opendatahub-io/odh-dashboard/pull/1497) for the above conclusion.

**Comment by pnaik1** [comment]
yeah I do agree with you @manaswinidas , but in this scenario , what is happening that the error is unknown until it reaches the graph tab (which is  in https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/concepts/pipelines/topology/core/PipelineVisualizationSurface.tsx ) 
to get what  you have suggested , I will need to move the error useState somewhere in pipelineDetails and pass it as props 
@andrewballantyne any suggestion on this?

**Comment by kywalker-rh** [comment]
I assume this is an error for both the graph and the yaml, correct? Is there anything a user can do to fix the issue, other than deleting the pipeline and uploading a new one? If there is an action the user can take to fix it we should make that clear and provide a link to allow them to do it. If there is nothing they can do to fix it, and they can only delete it, then we can include an action to delete it.

**Comment by andrewballantyne** [comment]
> I assume this is an error for both the graph and the yaml, correct? Is there anything a user can do to fix the issue, other than deleting the pipeline and uploading a new one? If there is an action the user can take to fix it we should make that clear and provide a link to allow them to do it. If there is nothing they can do to fix it, and they can only delete it, then we can include an action to delete it.

How do they fix it @kywalker-rh? Not in our UI obviously -- but if we just lock them down, say some vague error that is not directly related to their Pipeline (more about the Topology having rendering issues), shouldn't we give them the opportunity to see the error and view the yaml in one spot?

@manaswinidas great suggestion!

**Comment by kywalker-rh** [comment]
@andrewballantyne I was trying to understand the issue more. I don't have a full understanding of the possible error state, so its hard to prescribe something. If its a yaml issue, and we can provide them with a good error, and a way to view the yaml (even better if we can highlight the offending lines) then that is definitely the best option. If its more complicated than that maybe we should chat in a thread.

**Comment by andrewballantyne** [comment]
> @andrewballantyne I was trying to understand the issue more. I don't have a full understanding of the possible error state, so its hard to prescribe something. If its a yaml issue, and we can provide them with a good error, and a way to view the yaml (even better if we can highlight the offending lines) then that is definitely the best option. If its more complicated than that maybe we should chat in a thread.

Whoops, wrong ticket, deleted my other comment...

We have no idea where the issue is @kywalker-rh -- but we cannot render the graph. In this case, it was the name of a task not linking up to another task... so when the nodes were rendered the PF Topology was like "yeah, no that's invalid data" and this stops it from whitescreening.

But I think we want the YAML tab, so they can _try_ to make sense of it. If they can't, they are already in a bad place. It's a bad Pipeline YAML.

**Comment by andrewballantyne** [comment]
Okay, I'm getting all my wires crossed today 🤦 

Das was referencing moving away from the tabs (I thought she was saying we should add them lol)... no this is all good across the board... just going to approve this lol.

File: frontend/src/concepts/pipelines/topology/core/PipelineVisualizationSurface.tsx [hunk 1]
```diff
@@ -22,23 +24,39 @@ const PipelineVisualizationSurface: React.FC<PipelineVisualizationSurfaceProps>
...
+          nodes: renderNodes,
+          edges,
+        },
+        true,
+      );
+    } catch (error) {
+      if (error instanceof Error) {
+        setError(error);
+      }
+    }
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/topology/core/PipelineVisualizationSurface.tsx:47]: Can you maybe log something here if it's not somehow an instance of Error (which I imagine everything will be).

```suggestion
    } catch (error) {
      if (error instanceof Error) {
        setError(error);
      } else {
        // eslint-disable-next-line no-console
        console.error('Unknown error occurred rendering Pipeline Graph', error);
      }
    }
```

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Oh, one final request so we don't swallow the error and just end up in a state without ever being able to debug it.

**Review by Gkrumbach07 — APPROVED** [review]
Code looks good. waiting on cluster to test it live

/lgtm

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**